tests/scenario_tests/test_events_shared_channels.py

```diff
@@ -0,0 +1,514 @@
+import json
```

Comment:
This test suite is copied from `test_events.py` and then is modified to emulate shared channel situations.